packages/react-reconciler/src/ReactFiber.js

```diff
@@ -460,10 +460,13 @@ export function createFiberFromProfiler(
   const fiber = createFiber(Profiler, pendingProps, key, mode | ProfileMode);
   fiber.type = REACT_PROFILER_TYPE;
   fiber.expirationTime = expirationTime;
-  fiber.stateNode = {
-    duration: 0,
-    startTime: 0,
-  };
+  if (enableProfilerTimer) {
```

Comment:
We can shave off a bit by not even initializing `stateNode` for `Profiler` unless we're actually going to use it for timing.